### DISCUSSION TITLE
atomic_add cost in local vs global

### DISCUSSION THREAD

**Discussion by mikamyara** [pyopencl/pyopencl#693] [Q&A] [answered]
The question is in the title : I need to make lots of atomic_add in a global table (and I did not really find a way to avoid it, it's an algorithm close to hough transform, is someone has a better idea I will be happy to examine it !).
In between, I however think about the following : I can make some local tables and perform some atomic_adds inside, and this will lead to make fewer atomic_adds in the global space. Can it be better, about total computational time, than making all the atomic_adds directly in the global scope ?
Thanks a lot,
   Mikhaël

**Comment by inducer** [comment, marked as answer]
That's really got more to do with the OpenCL implementation and hardware you're using, and less with pyopencl. So without knowing those things, it's hard to even guess.

**Reply by mikamyara** [reply]
Ok I understand. I will ask in a more adapted forum. I am sorry I learn about all that through pyOpenCL, that's why I asked here.
Regards,
  Mike
